### DISCUSSION TITLE
How to update a table with a function

### DISCUSSION THREAD

**Discussion by brentrobbins** [supabase/supabase#4923] [Questions] [answered]
Is there a way to update my Profiles table when a user updates their email account (on the auth.users table) with a function? I have the insert function & trigger working, but I would like to make sure the email address is always synced/updated.

**Comment by brentrobbins** [comment, marked as answer]
Never mind, I found the answer within the "[Holiday Hackdays: Building an advent calendar - Day 4](https://www.youtube.com/watch?v=TijBVcV4jXw)" video.

**Reply by gunljunghusen** [reply]
I've tried to find it in the video but without luck, would you mind sharing the solution? Or perhaps a rough estimate of when they talk about it in the video?

**Reply by brentrobbins** [reply]
Hi @gunkarlsson You could create a function (use the updated email from auth.users to update your public.profiles table ) and trigger (when auth.user is updated) in Supabase. They don't do exactly this in the video, they insert the email when the user is created, but you can use that as a starting example. Let me know if that helps?

**Reply by filippobrigati** [reply]
Hello @brentrobbins, did you keep the same logic? why i can't update the profiles table when a user is changed

**Reply by brentrobbins** [reply]
Hi @filippo-brigati , would you mind letting me know what logic you are using and if you experiencing any errors?

**Reply by filippobrigati** [reply]
Yep, @brentrobbins, I followed the documentation and created the profiles table in which a row is added every time a user registers and this is my functions that return a trigger:

`
begin
    insert into public.profiles (id, username, email, created_by, role)
    values (new.id, new.raw_user_meta_data->>'username', new.email, new.raw_user_meta_data->>'company', CAST 
    (new.raw_user_meta_data->>'role' AS INTEGER));
    return new;
end;
`

this works correctly, I would like to implement a function that updates the table profiles when a user is updated (for example the email address is changed).

I tried to use the same logic then using the 'new' operator but it keeps giving me an error...

maybe i can't access the auth table after the update of the user, now i solved querying the table profiles after the correct response on auth edit query...I'm on self hosted supabase.

Thanks in advance.

**Reply by GaryAustin1** [reply]
You create an update after trigger and instead of doing insert in your function do update.  New should still work for the updated values (versus old for the existing row).  You might seek help in Supabase Discord.

**Reply by filippobrigati** [reply]
Thank you @GaryAustin1, But I have already tried, maybe I'm getting some values wrong, I'll also check on discord, I'll write here if I can solve it.

Thanks again

**Reply by quebrandolink** [reply]
Conseguiu resolver @filippo-brigati ?

**Reply by filippobrigati** [reply]
@quebrandolink not yet, I remedied this by making a normal .update() call, but today I installed a new version of supabase so in the next few days I'll try again using that method, I'll let you know. 🙂

please remind me if I don't answer you.
